src/modules/tree/DynamicTreeView.tsx

```diff
@@ -24,37 +28,82 @@ type Props = {
   onTreeItemSelect?: (value: string) => void;
   isLoading?: boolean;
   onlyShowContainerItems?: boolean;
+  mainItem?: DiscriminatedItem;
 };
 
+interface NodeProps {
+  element: INode;
+  isBranch: boolean;
+  isExpanded: boolean;
+  isSelected: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getNodeProps: () => any;
+  onSelect: (id: UUID) => void;
+}
+const RenderedNode = ({
+  element,
+  isBranch,
+  isExpanded,
+  getNodeProps,
+  onSelect,
+  isSelected,
+}: NodeProps) => (
+  <button
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    {...getNodeProps()}
+    type="button"
+    className={`${buildTreeItemClass(element.id as string)} flex-center`}
+  >
+    {isBranch &&
+      (isExpanded ? (
+        <IconButton sx={{ padding: 0 }}>
+          <ExpandMoreIcon />
+        </IconButton>
+      ) : (
+        <IconButton sx={{ padding: 0 }}>
+          <ChevronRightIcon />
+        </IconButton>
+      ))}
+    <Typography
+      component="button"
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+        gap: '4px',
+        background: isSelected ? '#e0e2e2' : 'none',
+      }}
+      onClick={() => onSelect(element?.id as UUID)}
+    >
+      <FolderIcon fontSize="small" />
+      {element.name}
+    </Typography>
+  </button>
+);
+
 const DynamicTreeView = ({
   id,
   header,
   items,
-  initialExpendedItems = [],
-  selectedId,
   onTreeItemSelect,
   isLoading = false,
   onlyShowContainerItems = true,
+  mainItem,
```

Comment:
I think it's too messy to either input the `mainItem` or the `items`. Can't you use only `items` with `[mainItem]`?

Reply:
I tried to have it this way, But I'm counting on `extra.folder.childrenOrder` to handle item children, and useItem unfortunately doesn't return this data, that's why I handle it this way